lib/chef/application.rb

```diff
@@ -134,7 +134,9 @@ def set_specific_recipes
     # that a user has configured a log_location in client.rb, but is running
     # chef-client by hand to troubleshoot a problem.
     def configure_logging
-      Chef::Log.init(MonoLogger.new(Chef::Config[:log_location]))
+      if Chef::Config[:log_location] || Chef::Log.unconfigured?
```

Comment:
should this be an `&&` rather than `||`? It's not clear how the second branch would ever be hit. Please can you add some tests for this.